rust/processor/src/processors/econia_processor.rs

```diff
@@ -0,0 +1,212 @@
+        let market_registration_type =
+            format!("{}::registry::MarketRegistrationEvent", self.econia_address);
+        let mut market_registration_events = vec![];
+        for transaction in transactions {
+            let (txn, _, events, _, _) = TransactionModel::from_transaction(&transaction);
+            for (index, event) in events.iter().enumerate() {
+                if event.type_ != market_registration_type {
+                    continue;
+                }
+                let txn_version = BigDecimal::from(txn.version);
+                let event_idx = BigDecimal::from(index as u64);
+                let market_id = opt_value_to_big_decimal(event.data.get("market_id"));
+                let lot_size = opt_value_to_big_decimal(event.data.get("lot_size"));
+                let tick_size = opt_value_to_big_decimal(event.data.get("tick_size"));
+                let min_size = opt_value_to_big_decimal(event.data.get("min_size"));
+                let underwriter_id = opt_value_to_big_decimal(event.data.get("underwriter_id"));
+                let time = *block_height_to_timestamp
+                    .get(&event.transaction_block_height)
+                    .unwrap();
+                let base_name_generic;
+                let base_account_address;
+                let base_module_name;
+                let base_struct_name;
+                if value_is_empty_string(&event.data["base_name_generic"]) {
+                    base_name_generic = None;
+                    base_account_address = Some(String::from(
+                        // cannot panic because account_address always exists as a string
+                        // assuming base_name_generic was empty
+                        event.data["base_type"]["account_address"].as_str().unwrap(),
+                    ));
+                    let base_module_name_hex =
+                        // cannot panic because module_name always exists as a string
+                        // assuming base_name_generic was empty
+                        event.data["base_type"]["module_name"].as_str().unwrap();
+                    let base_struct_name_hex =
+                        // cannot panic because struct_name always exists as a string
+                        // assuming base_name_generic was empty
+                        event.data["base_type"]["struct_name"].as_str().unwrap();
+                    base_module_name = Some(hex_to_string(base_module_name_hex));
+                    base_struct_name = Some(hex_to_string(base_struct_name_hex));
+                } else {
+                    base_name_generic = Some(String::from(
+                        // cannot panic because base_name_generic always exists as a string
+                        event.data["base_name_generic"].as_str().unwrap(),
+                    ));
+                    base_account_address = None;
+                    base_module_name = None;
+                    base_struct_name = None;
+                }
```

Comment:
Here is a more concise way of doing this:

```suggestion
                let (base_name_generic, base_account_address, base_module_name, base_struct_name) =
                    if value_is_empty_string(&event.data["base_name_generic"]) {
                        (
                            None,
                            Some(event.data["base_type"]["account_address"].as_str()?),
                            Some(hex_to_string(event.data["base_type"]["module_name"].as_str()?)?),
                            Some(hex_to_string(event.data["base_type"]["struct_name"].as_str()?)?),
                        )
                    } else {
                        Some(String::from(event.data["base_name_generic"].as_str()?)),
                        None,
                        None,
                        None,
                    };
```

Also replaced `unwrap` with `?`. As you say, this should not be a problem, but `?` is shorter, and if there was to be a problem (let's say, the data we got was corrupted or changed somehow), it would be properly handled.

Reply:
Thanks for the suggestion--it appears that the `as_str` calls are returning optional strings rather than results. I guess it's kind of confusing that options and results both have unwrap.